ui/components/auth-required.js

```diff
@@ -4,14 +4,21 @@ import useSWR from 'swr'
 export default function AuthRequired({ children }) {
   const { data: auth, error } = useSWR('/api/users/self')
   const router = useRouter()
+  const { asPath } = router
+  const logout = window.localStorage.getItem('logout')
 
   if (!auth && !error) {
     return null
   }
 
   if (!auth?.id) {
-    router.replace('/login')
-    return null
+    if (!logout && asPath !== '/destinations' && asPath !== '/') {
+      router.replace(`/login?next=${asPath.slice(1)}`)
```

Comment:
Do we need to slice here?

Also, I think we'd need to escape `asPath`, e.g:

```router.replace(`/login?next=${encodeURIComponent(asPath)}`)```

Reply:
yes, we need to slice to remove the `/`